From 3ade4bc736a4f6c9246fdf3e5f388e182b94fbe5 Mon Sep 17 00:00:00 2001
From: Johannes Demel <demel@ant.uni-bremen.de>
Date: Mon, 15 Jun 2020 23:14:26 +0200
Subject: [PATCH 7/9] fix: Adjust min/max for 32f_s32f_convert_8i kernel

This kernel used `CHAR_MIN` to determine the minimum value which may be
0. It is expected that this value is -127. Thus, we move to `INT8_MIN`.

Fix #390
---
 kernels/volk/volk_32f_s32f_convert_8i.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernels/volk/volk_32f_s32f_convert_8i.h b/kernels/volk/volk_32f_s32f_convert_8i.h
index 242c3bd..ea16ef8 100644
--- a/kernels/volk/volk_32f_s32f_convert_8i.h
+++ b/kernels/volk/volk_32f_s32f_convert_8i.h
@@ -78,8 +78,8 @@
 
 static inline void volk_32f_s32f_convert_8i_single(int8_t* out, const float in)
 {
-    float min_val = CHAR_MIN;
-    float max_val = CHAR_MAX;
+    float min_val = INT8_MIN;
+    float max_val = INT8_MAX;
     if (in > max_val) {
         *out = (int8_t)(max_val);
     } else if (in < min_val) {
-- 
2.20.1

